eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Snap.hs

```diff
@@ -80,16 +83,14 @@ instance EraTxOut era => STS (ShelleySNAP era) where
 -- where important changes were made to the source code.
 snapTransition ::
   forall era.
+  EraPParams era =>
   TransitionRule (ShelleySNAP era)
 snapTransition = do
-  TRC (lstate, s, _) <- judgmentContext
+  TRC (snapEnv, s, _) <- judgmentContext
 
-  let LedgerState
-        (UTxOState _utxo _ fees _ incStake)
-        (DPState dstate pstate)
-        _ = lstate
-      -- stakeSnap = stakeDistr @era utxo dstate pstate  -- HISTORICAL NOTE
-      istakeSnap = incrementalStakeDistr @(EraCrypto era) incStake dstate pstate
+  let SnapEnv (LedgerState (UTxOState _utxo _ fees _ incStake) (DPState dstate pstate) _) pp = snapEnv
+      -- stakeSnap = stakeDistr @era utxo dstate pstate  -EraCrypto- HISTORICAL NOTE
```

Comment:
:thinking: how can we make this clearer? (I realize you did not introduce it...). maybe:

> This is equivalent to what is written in the spec, namely `stakeSnap = stakeDistr @era utxo dstate pstate`

?

Reply:
Yes good idea, a few times I deleted this before reading the `HISTORICAL NOTE` end,  much better to write a human explanation like you proposed